src/net_setup.c

```diff
@@ -486,18 +484,30 @@ bool setup_myself_reloadable(void) {
 		case PROXY_EXEC:
 			if(!space || !*space) {
 				logger(DEBUG_ALWAYS, LOG_ERR, "Argument expected for proxy type exec!");
+				free(proxy);
 				return false;
 			}
 
-			proxyhost =  xstrdup(space);
+			free(proxyhost);
+			proxyhost = xstrdup(space);
 			break;
 
 		case PROXY_SOCKS4:
 		case PROXY_SOCKS4A:
 		case PROXY_SOCKS5:
 		case PROXY_HTTP:
+			free(proxyhost);
```

Comment:
I think all those `free()`s and setting to `NULL` should be done once before the `switch`-statement.